weather.js

```diff
@@ -0,0 +1,70 @@
+const axios = require("axios");
+const dotenv = require('dotenv');
+dotenv.config();
+
+const apiURLPrefix = "http://api.openweathermap.org/";
+
+/**
+ * Runs a conditional to distinguish if Zipcode or address string was entered.
+ * @param {string|number}input
+ * @returns {Promise<void>}
+ */
+async function getWeather(input) {
+    if (isNaN(input)) {
+        const data = await getCoordinates(input);
+        await getWeatherData(`lat=${data[0]["lat"]}&lon=${data[0]["lon"]}`);
+        return;
+    }
+    await getWeatherData(`zip=${input}`);
+}
+
+/**
+ * Gets coordinates for city/state entry.
+ * @param {string} input City,State
+ * @returns {array[object]}
+ */
+async function getCoordinates(input) {
+    const geoURL = `${apiURLPrefix}geo/1.0/direct?q=${input},us&appid=${process.env.W_API}`;
+    let data = await buildResponseBody(geoURL);
+    if (data.length === 0) {
+        return console.error(`${input} is not a valid location, please try again.`);
```

Comment:
Throw an exception rather than `console.error` here.
You can catch that exception where this method is used and handle it accordingly (console.error at that point).
That allows this method to be more reusable accounting for various context specific ways of handling things.